base/condition.jl

```diff
@@ -79,7 +79,6 @@ lock(f, c::GenericCondition) = lock(f, c.lock)
 
 # have waiter wait for c
 function _wait2(c::GenericCondition, waiter::Task)
```

Comment:
Instead of #42977, why not bring back `ct`?

```suggestion
function _wait2(c::GenericCondition, waiter::Task)
    ct = current_task()
```

Reply:
https://github.com/JuliaLang/julia/pull/42977#issuecomment-962528766

Reply:
Thanks I get what this was about now.

The perils of having PRs sitting around for months — seems I forgot about this cleanup (which is now a bug).

I just reverted this part to put `ct` back in. It's unrelated to the changes here anyway.